frontend/libraries/ballerina-core/src/forms/domains/singleton/template.tsx

```diff
@@ -26,14 +26,15 @@ export const Form = <Entity, FieldStates, Context, ForeignMutationsExpected>() =
               ({
                 ...props.foreignMutations,
                 onChange: (_: BasicUpdater<Entity[field]>, path) => {
-                  props.setState(_ => ({ ..._, 
+                  if(validation) {props.setState(_ => ({ ..._,
                     modifiedByUser: true, 
                     validation:Debounced.Updaters.Template.value<FormValidatorSynchronized>(Synchronized.Updaters.value(replaceWith(unit)))(_.validation),
                     [field]:({
                       ..._[field], 
                       modifiedByUser:true,
                       validation:Debounced.Updaters.Template.value<FormValidatorSynchronized>(Synchronized.Updaters.value(replaceWith(unit)))(_[field].validation),
                     }) }))
+                  }
```

Comment:
does this if make sense to you? we only run the validation if defined (and hence only set state of modified by user if there's validation) cuts a render out.